<launch>
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="-2.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>

  <arg name="freq" default="30" />
  <arg name="avoid_angles" default="10" doc="# sensor divisions to count as straight ahead, one sided"/>
  <arg name="min_dist" default="0.75" doc="Minimum distance to obstacle before stop+turn"/>

  <arg name="rec" default="false" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/turtlebot3_world.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <param name="freq_set" type="int" value="$(arg freq)" />
  <param name="avoid_angles_set" type="int" value="$(arg avoid_angles)" />
  <param name="min_avoid_distance" type="double" value="$(arg min_dist)" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />

  <node pkg="elyard_walker" name="cmd_algo" type="walker_script" />

  <node pkg="rosbag" type="record" name="rosbag_record" args="--all --exclude=/camera/* --duration=30" output="screen" if="$(arg rec)" />

</launch>
